radicle-crypto/src/ssh/keystore.rs

```diff
@@ -0,0 +1,195 @@
+use std::os::unix::fs::DirBuilderExt;
+use std::path::{Path, PathBuf};
+use std::{fs, io};
+
+use thiserror::Error;
+
+use crate::{KeyPair, PublicKey, SecretKey, Signature, Signer, SignerError};
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error(transparent)]
+    Io(#[from] io::Error),
+    #[error("ssh keygen: {0}")]
+    Ssh(#[from] ssh_key::Error),
+    #[error("invalid key type, expected ed25519 key")]
+    InvalidKeyType,
+    #[error("keystore already initialized")]
+    AlreadyInitialized,
+}
+
+/// Stores keys on disk, in OpenSSH format.
+#[derive(Debug)]
+pub struct Keystore {
+    path: PathBuf,
+}
+
+impl Keystore {
+    /// Create a new keystore pointing to the given path. Use [`Keystore::init`] to initialize.
+    pub fn new<P: AsRef<Path>>(path: &P) -> Self {
+        Self {
+            path: path.as_ref().to_path_buf(),
+        }
+    }
+
+    /// Get the path to the keystore.
+    pub fn path(&self) -> &Path {
+        self.path.as_path()
+    }
+
+    /// Initialize a keystore by generating a key pair and storing the secret and public key
+    /// at the given path.
+    ///
+    /// The `comment` is associated with the private key.
+    /// The `passphrase` is used to encrypt the private key.
+    pub fn init(&self, comment: &str, passphrase: &str) -> Result<PublicKey, Error> {
```

Comment:
You should probably use [secstr](https://docs.rs/secstr/0.5.1/secstr/struct.SecUtf8.html) for passphrases

Reply:
Yeah, is there a benefit over `Zeoizing<String>`? I'm going to tackle all the zeroizing stuff in a separate PR.

Reply:
I'm not sure, but `secstr` seems to cover a lot of scenarios:
> A data type suitable for storing sensitive information such as passwords and private keys in memory, featuring constant time equality, mlock and zeroing out.

Reply:
Added issue #79